### ISSUE TITLE
A better volume manager int the drop-down menu

### ISSUE THREAD

**Issue by mymike00** [ubports/ubports-touch#81]
Would be very useful if I can manage more than one volume in the drop-down menu (obviously under the Volume tab). Something like [uVolMan](https://openstore.ubports.com/app/uvolman.dfiloni), or just to control the Multimedia volume because to change it I have to use it (play music, open a game, ...) and it's very uncomfortable.
Something like this:
![volume manager](https://cloud.githubusercontent.com/assets/28359593/26760313/375c708c-4915-11e7-866e-05ae34dc86ad.png)

**Comment by Flohack74** [comment]
I know its a pain to have only one volume... E.g.:

 I am on the phone with someone that I barely understand, I turn up the volume to 100%.

Next thing happens is a call comes in and I get a heart attack because the ringtone is screaming and wakes up the dead. :)

**Comment by Stefano0101** [comment]
Yes, this would be very good if implemented like on Android. It's clearly a problem for other user as well.
It was reported back on launchpad as well, never fixed. uVolMan is the solution at the moment.
Meizu Pro 5.

**Comment by mymike00** [comment]
> I know its a pain to have only one volume...

But there are already two different volumes, the thing is there is only one controller.
If I don't have any app open and I raise up the volume, only the Ringtone & Notifications volume will change. Otherwise, if I'm listening to the music and I raise up the volume only the Multimedia one will change: if I want I can keep Multimedia volume to zero but when someone call me I can hear the ringtone.

**Comment by ernesst** [comment]
The same while listening with an headset...the volume doesn't go down to a safety level when i plug it and i kill my ear...
Bluetooth in my car full level, headset at work level very low...

**Comment by mymike00** [comment]
> The same while listening with an headset...the volume doesn't go down to a safety level when i plug it and i kill my ear...
Bluetooth in my car full level, headset at work level very low...

But that is another problem: you use only with the Multimedia volume and the problem is between bluetooth and headset...

**Comment by ChloeWolfieGirl** [comment]
I dont know if this concerns having 3 sliders like android 
- Notifications
- Media
- Ringtones

Or if it concerns different applications, mostly because if you have music playing and you go and decide to play some gameboy, the gameboy music and the music from your music app play at once, luckily there is an option in the app to mute the music, but it would be nice if the app developer didn't have to worry about that.

**Comment by mymike00** [comment]
well, I thought with only 2 sliders (notifications and ringtones are actually one whole volume) to make things easier and because I don't understand why different volume are better then one...
while for the second thing you said I think every app should manage its volume by itself, but, as you said, if you want to listen to music while playing isn't immediate: why don't put a ChechBox/Switch to enable the sound only for the app which is reproducing something? for example: I'm listening to music then I tap the checkbox/switch and now only the music app can reproduce multimedia sounds (ringtone and notifications still enabled) but other apps can't, like if you want to play a game, it won't reproduce anything because you allowed only the music app to reproduce multimedia stuff...

idk if I'm clear...

**Comment by Flohack74** [comment]
Hmm not really clear what are the proposed ideas atm ;) 

Let me tell from the technical background that there is a sound subsystem in the OS which will receive PCM data to be played. Each App can modify the amplitude of this PCM data from 0 to 100%. So that covers the in-App volume sliders, YouTube bars etc.

The next step is the sound subsystem, which can now decide what to do with this PCM data from all the applications it receives. It can just mix them up, send it to the output DAC, and provide 1 volume control for each output (that would mean in our case separate level for headphones, phone earpiece and speaker.

Or, it could act like Windows does now recently: Provide 1 slider for each Application and let them be set separatly.

Or 1 slider by "intent" (Notification, Ringtone, Media, Alerts, ...)

**Comment by Flohack74** [comment]
See here:
![grafik](https://user-images.githubusercontent.com/16186470/27931628-1d693972-629b-11e7-969d-dc7a861b975e.png)

**Comment by Flohack74** [comment]
IMHO it is not necessary to be able to control each App individually. But I like the idea to do it by groups/intents of sounds. (I like the word intent, it was wisely chosen by Android guys xD)

**Comment by ernesst** [comment]
I think 3 slider should enough
like 
![2017-07-06 16_36_05-presentation1 - powerpoint](https://user-images.githubusercontent.com/11337464/27934057-9b27961a-6269-11e7-88c5-02583610004b.png)

**Comment by Flohack74** [comment]
@ernesst is this mockup or actual running code? =)

**Comment by ernesst** [comment]
A mockup

**Comment by mymike00** [comment]
My idea is to put a checkbox/switch near the media player in the top edge menu.
When an app is playing sone sound, like the music app, you can enable this chechbox/switch so the app that is playing before could be the only app to reproduce multimedia audio. This could be helpful when touch want listening to music while playing another game that has some sounds.
But the idea of the volume mixer or the groups of @Flohack74

**Comment by ChloeWolfieGirl** [comment]
Perhaps the media slider could have a dropdown box, which would allow you to change individual apps audio? That way people would have more advance audio options if they wished, while also having it simplistic enough for those who dont care for deeper audio options.

**Comment by mateosalta** [comment]
I always wanted a master mute, I could be in silent mode with the "volume" slider all th way down just to have it blast youtube or random browser multimedia...

**Comment by NeoTheThird** [comment]
Moved to 15.04 OTA-3 due to time constraints.

**Comment by ChloeWolfieGirl** [comment]
Just to throw some extra chat into this, when you're in a call, android goes from 3 sliders to 4, as it adds the call as an extra volume slider.

On android it goes
- Notifications
- Multimedia
- Alarms

then it adds one to the top when you're in call
- Call
- Notifications
- Multimedia
- Alarms

**Comment by Flohack74** [comment]
Is the Ringtone on notifications or alarms? Xd

**Comment by ChloeWolfieGirl** [comment]
@Flohack74 Notifications.

![photo_2017-08-04_19-40-43](https://user-images.githubusercontent.com/8162805/28982367-752e1736-794d-11e7-8461-f3bae0ab2ab7.jpg)
![photo_2017-08-04_19-43-23](https://user-images.githubusercontent.com/8162805/28982368-753708a0-794d-11e7-924e-bd3f71b42286.jpg)

**Comment by mihaelmilea** [comment]
I am for notifications and ringtone together in one slider. Why would we need different controls for them?

**Comment by mihaelmilea** [comment]
Also, I believe it would be great if the mute button would mute everything, not only the ringtone and notifications. For example, if the phone is on mute, right now I still get sound from a video within a webpage (not good). That means that right now there in no way to mute the sound from a webpage except only after it makes a lot of noise! - or whatever the volume level was setup the last time I used that specific app.

**Comment by mymike00** [comment]
@mihaelmilea you can use uVolMan right now. It's an app that lets you control all the different volumes

**Comment by Flohack74** [comment]
uVolMan das not really work well on the E5, plus we want to have this integrated in the indicator. So the ticket is still valid.

**Comment by mihaelmilea** [comment]
Thank you @mymike00, yes, I am using for now the uVolMan. Thank you @Flohack74 for still keeping this alive! Could then the mute switch mute all the volumes (notifications, ringtone as well as multimedia)? - except maybe the call volume...

**Comment by mymike00** [comment]
> uVolMan das not really work well on the E5, plus we want to have this integrated in the indicator. So the ticket is still valid.

on E4.5 it works fine, I think... I use it rarely but I didn't see problems. I suggested it for now while the indicator can't do it.
And I know we want it in the indicator as well I know the ticket is still valid given that I open this issue ^^

**Comment by mauricioduarte01** [comment]
@mateosalta Looking great! I hate now hurting my ears when using headphones because of the bad volume config is made

**Comment by mymike00** [comment]
@mateosalta when di they published it? and what do they refer to as before? it's completely different than now...
and what is it the smaller screenshot near each image?

**Comment by mateosalta** [comment]
A weak ago, the design team from canonical released documents they were working on: https://community.ubuntu.com/t/mockups-new-design-discussions/1898/185

The smaller picture, is how it would look in a larger desktop/convergence, rather than the full-screen drop-down on phone, and sidebar on tablet.

**Comment by mihaelmilea** [comment]
I don't know if this is the opportune place to add this or I should start a new issue but I was thinking about different call volume settings for phone speaker, loud speaker and headphones.

Because when I use the phone on loud speaker I need the volume at maximum but when I plug the headphones I would like the volume to be adjusted automatically to what I previously set which is a lower volume.

Right now, when I adjust the volume during a call, it adjusts the volume for all the three setups: loud speaker, phone speaker and headsets.

And, if we think about this, alarms and ringtones shoud sound at a lower / different volume in the headsets.

**Comment by mihaelmilea** [comment]
Yes, call me paranoic :) I am afraid of hidden volumes that I forget to check and then I don't hear the alarm for example. Is it too much if we have all sliders displayed to see and set? I dare to humbly submit a mockup. Then, when the headphones are plugged in, the values should be different. And when a phonecall is switched to loud speaker, the volume for that should be different then the phonecall volume levels set for phone speaker or headsets. Then we can also see what the Mute button does :) - for example, right now, it doesn't mute the multimedia volume. My suggestion would be to mute Ringtones, Multimedia and Notifications.

![volume_sliders_mockup](https://user-images.githubusercontent.com/30057613/50725509-4dd8ae00-1107-11e9-94d4-faf811579d20.png)

**Comment by Stefano0101** [comment]
That looks good. Surely something like this is needed for UT.
